scripts/dockerbot/check_prs.py

```diff
@@ -277,9 +313,19 @@ def check_status(pr):
                 logger.info('Setting task {} for pr {} to {}'
                             .format(task['stream'], pr['number'], states[task['result']]))
                 set_status(commit, states[task['result']], task['stream'])
+            task_states[task['stream']] = states[task['result']]
```

Comment:
I think this is still in the wrong place, as far as I remember, check_status is run against each PR, we then get the latest run_id, and iterate through all the tasks for the latest run_id. During this iteration line 316 is going to be hit each time we iterate the task list. Even if the break is hit in 305, it's only going to break out of the loop at 302, which I _think_ means we are still goign to have task_states be populated. Even if the "pending" etc are filtered out, we will still get "failure" everytime this is run.

Unless I'm missing something. Which could be the case ;)